packages/fluxible-router/lib/createNavLinkComponent.js

```diff
@@ -132,7 +134,7 @@ module.exports = function createNavLinkComponent (overwriteSpec) {
                 return;
             }
 
-            if (href[0] !== '/') {
+            if (href[0] !== '/' || (this.props.validate && !routeStore.getRoute(href))) {
```

Comment:
Should we move `(this.props.validate && !routeStore.getRoute(href))` check to line 150 to be:
```
if (this.props.validate && !routeStore.getRoute(href)) {
    return;
}
```

Line 148 will make sure absolute urls with the same origin with be converted to relative url, which makes it perfect for matching route.